solutions/perimeter/fortigate/fortigate-ap-primary.yaml

```diff
@@ -0,0 +1,391 @@
+## Active Passive Cluster - Primary Node
+---
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeInstance
+metadata:
+  name: perimeter-fgt-primary-instance
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: perimeter-project-id # kpt-set: ${perimeter-project-id}
+spec:
+  resourceID: fgt-primary-instance
+  description: Fortigate Primary Instance
+  machineType: n2-standard-4
+  zone: northamerica-northeast1-a
+  canIpForward: true
+  scheduling:
+    automaticRestart: true
+    onHostMaintenance: MIGRATE
+    preemptible: false
+  serviceAccount:
+    serviceAccountRef:
+      name: perimeter-fortigatesdn-sa
+    scopes:
+    - cloud-platform
+  
+  # TODO: Schedule
+  # resourcePolicies:
+  # - name: TBD
+  
+  # Create a new boot disk from an image
+  bootDisk:
+    autoDelete: true
+    deviceName: persistent-disk-0
+    initializeParams:
+      type: pd-ssd
+      sourceImageRef:
+        external: image-path # kpt-set: ${fgt-primary-image}
+  
+  # Logging Disk from existing resource
+  attachedDisk:
+  - sourceDiskRef:
+      name: perimeter-fgt-primary-log-disk
+  
+  networkInterface:
+  # External
+  - name: nic0
+    networkRef:
+      name: perimeter-global-external-vpc
+    subnetworkRef:
+      name: perimeter-nane1-external-paz-snet
+    networkIpRef:
+      name: perimeter-fgt-primary-ext-address
+      kind: ComputeAddress
+  # Internal
+  - name: nic1
+    networkRef:
+      name: perimeter-global-internal-vpc
+    subnetworkRef:
+      name: perimeter-nane1-internal-paz-snet
+    networkIpRef:
+      name: perimeter-fgt-primary-int-address
+      kind: ComputeAddress
+  # Transit
+  - name: nic2
+    networkRef:
+      name: perimeter-global-transit-vpc
+    subnetworkRef:
+      name: perimeter-nane1-transit-paz-snet
+    networkIpRef:
+      name: perimeter-fgt-primary-transit-address
+      kind: ComputeAddress
+  # MGMT
+  - name: nic3
+    networkRef:
+      name: perimeter-global-mgmt-vpc
+    subnetworkRef:
+      name: perimeter-nane1-mgmt-rz-snet
+    networkIpRef:
+      name: perimeter-fgt-primary-mgmt-address
+      kind: ComputeAddress
+
+  tags:
+  - internet-egress-route
+
+  # Metadata to enable serial console and bootstrap FGT
+  metadata:
+  - key: serial-port-enable
+    value: "TRUE"
+  - key: license
+    value: LICENSE # kpt-set: ${fgt-primary-license}
+  - key: user-data
+    value: |
+      config system global
+          set hostname "fgt-ap-primary"
+          set admintimeout 60
+          set timezone 12
+      end
+      config system admin
+          edit "admin"
+              set password fortinet
```

Comment:
@CraigBaltzer @amcmullin  :  Our friend @cartyc is looking into a way to remove this hardcoded secret out of this file and maybe have it inside an external secret manager solution. Remember that this password is used only to bootstrap the appliances and that the Platform Admin should be changing it as part of the "day 2" operation

Reply:
@davelanglois-ssc, @amcmullin understood that its on the day 2 list to change, concern is an interrupted process where the VM gets deployed but there is a significant lag until the "day 2" work is done if the process isn't automated in a pipeline. 

The current password is poor as it would be within the first 3 or 4 guesses someone would make trying to get in without any access to the yaml. 

Storing a "strong" password in an external secret manager is a good option. 

Anything that can be done to prevent access until day 2 operations are underway would help mitigate exposure, such as powered off the VMs immediately after deployment (assuming rights to power on VMs is tightly controlled), limiting FortiGate VM console/UI access to a small number of hosts, etc.

Reply:
For this round can we at least change the admin password to something that isn't easily guessable?